### PR TITLE
[JENKINS-34745] - Prevent CheckUpdates PeriodicWork death if update site cert is missing

### PR DESCRIPTION
**Pull request by oleg-nenashev**
https://issues.jenkins-ci.org/browse/JENKINS-34745
Special thanks to @lanwen for https://github.com/yandex-qatools/juseppe, which allowed to catch it.

@reviewbybees @jenkinsci/code-reviewers

### CODE REVIEW

File: core/src/main/java/hudson/PluginManager.java [hunk 1]
```diff
@@ -1406,20 +1406,24 @@ public HttpResponse doUploadPlugin(StaplerRequest req) throws IOException, Servl
     @Restricted(NoExternalUse.class)
     @RequirePOST public HttpResponse doCheckUpdatesServer() throws IOException {
         Jenkins.getInstance().checkPermission(Jenkins.ADMINISTER);
-        for (UpdateSite site : Jenkins.getInstance().getUpdateCenter().getSites()) {
-            FormValidation v = site.updateDirectlyNow(DownloadService.signatureCheck);
-            if (v.kind != FormValidation.Kind.OK) {
-                // TODO crude but enough for now
-                return v;
+        try {
```
Comment (oleg-nenashev) [inline core/src/main/java/hudson/PluginManager.java]: this code just adds try/catch just in case a RuntimeException happens in other place

**Comment by andresrc** [comment]
:bee:

**Comment by varmenise** [comment]
🐝

File: core/src/main/java/jenkins/util/JSONSignatureValidator.java [hunk 2]
```diff
@@ -82,7 +82,11 @@ public FormValidation verifySignature(JSONObject o) throws IOException {
 
             // this is for computing a signature
             Signature sig = Signature.getInstance("SHA1withRSA");
-            sig.initVerify(certs.get(0));
+            if (certs.isEmpty()) {
+                return FormValidation.error("No certificate found in " + name + ". Canot verify the signature");
+            } else {
```
Comment (varmenise) [inline core/src/main/java/jenkins/util/JSONSignatureValidator.java]: Typo `s/Canot/Cannot`

**Comment by varmenise** [comment]
just a typo

**Comment by oleg-nenashev** [comment]
@reviewbybees done

File: core/src/main/java/jenkins/util/JSONSignatureValidator.java [hunk 3]
```diff
@@ -82,7 +82,11 @@ public FormValidation verifySignature(JSONObject o) throws IOException {
 
             // this is for computing a signature
             Signature sig = Signature.getInstance("SHA1withRSA");
-            sig.initVerify(certs.get(0));
+            if (certs.isEmpty()) {
+                return FormValidation.error("No certificate found in " + name + ". Cannot verify the signature");
```
Comment (lanwen) [inline core/src/main/java/jenkins/util/JSONSignatureValidator.java]: use `FormValidation.error("No certificate found in %s. Cannot verify the signature", name)` style as more readable

File: core/src/main/java/hudson/PluginManager.java [hunk 4]
```diff
@@ -1406,20 +1406,24 @@ public HttpResponse doUploadPlugin(StaplerRequest req) throws IOException, Servl
...
+            for (UpdateSite site : Jenkins.getInstance().getUpdateCenter().getSites()) {
```
Comment (KostyaSha) [inline core/src/main/java/hudson/PluginManager.java]: `activeInstance()` if you are already touching it?

Comment (oleg-nenashev) [inline core/src/main/java/hudson/PluginManager.java]: There was https://github.com/jenkinsci/jenkins/pull/2297

**Comment by oleg-nenashev** [comment]
I suppose it's fine now

**Pull request merged**